lib/scripts/k2s/system/upgrade/Start-ClusterUpgrade.ps1

```diff
@@ -81,17 +79,21 @@ function Start-ClusterUpgrade {
         [parameter(Mandatory = $false, HelpMessage = 'HTTP proxy if available')]
         [string] $Proxy,
         [Parameter(Mandatory = $false, HelpMessage = 'Skip takeover of container images')]
-        [switch]
-        $SkipImages = $false
+        [switch] $SkipImages = $false,
+        [parameter(Mandatory = $false, HelpMessage = 'Directory for resource backup')]
+        [string] $BackupDir = ''
     )
     $errUpgrade = $null
     $addonsBackupPath = $null
     $logFilePathBeforeUninstall = $null
-    $tpath = $null
     $coresVM = $null
     $memoryVM = $null
     $storageVM = $null
 
+    if ($BackupDir -eq '') {
```

Comment:
Does it makes sense to check if BackupDir really exists when passed by user?. If not throw before upgrade procedure starts.

Reply:
Default value of BackupDir is empty string even if the user does not specify a path. Then the temp dir should be used

Reply:
I mean, if the user passes invalid path, do we check that?

Reply:
yes you're right, this is missing